services/worker/src/worker/job_runners/split_names_from_dataset_info.py

```diff
@@ -0,0 +1,131 @@
+        )
+
+
+class PreviousStepStatusError(SplitNamesFromDatasetInfoJobRunnerError):
+    """Raised when the previous step gave an error. The job should not have been created."""
+
+    def __init__(self, message: str, cause: Optional[BaseException] = None):
+        super().__init__(message, HTTPStatus.INTERNAL_SERVER_ERROR, "PreviousStepStatusError", cause, False)
+
+
+class PreviousStepFormatError(SplitNamesFromDatasetInfoJobRunnerError):
+    """Raised when the content of the previous step has not the expected format."""
+
+    def __init__(self, message: str, cause: Optional[BaseException] = None):
+        super().__init__(message, HTTPStatus.INTERNAL_SERVER_ERROR, "PreviousStepFormatError", cause, False)
+
+
+class SplitNameItem(TypedDict):
+    dataset: str
+    config: str
+    split: str
+
+
+class SplitNamesFromDatasetInfoResponseContent(TypedDict):
+    split_names: List[SplitNameItem]
+
+
+def compute_split_names_from_dataset_info_response(
+    dataset: str,
+    config: str,
+) -> SplitNamesFromDatasetInfoResponseContent:
+    """
+    Get the response of /split-names-from-dataset-info for one specific dataset and config on huggingface.co
+    computed from cached response in /dataset-info step.
+
+    The /split-names-from-dataset-info response generated by this function does not include stats about the split,
+    like the size or number of samples. See /dataset-info or /sizes for that.
+
+    Args:
+        dataset (`str`):
+            A namespace (user or an organization) and a repo name separated
+            by a `/`.
+        config (`str`):
+            A configuration name.
+    Returns:
+        `SplitNamesFromDatasetInfoResponseContent`: An object with the list of split names for the dataset and config.
+    <Tip>
+    Raises the following errors:
+        - [`~job_runners.split_names_from_dataset_info.PreviousStepStatusError`]
+          If the the previous step gave an error.
+        - [`~job_runners.split_names_from_dataset_info.PreviousStepFormatError`]
+            If the content of the previous step has not the expected format
+        - [`~libcommon.dataset.DatasetNotFoundError`]
+            If previous step content was not found for the dataset
+    </Tip>
+    """
+    logging.info(f"get split names from dataset info for dataset={dataset}, config={config}")
+    try:
+        response = get_response(kind="/parquet-and-dataset-info", dataset=dataset)
+    except DoesNotExist as e:
+        raise DatasetNotFoundError("No response found in previous step for this dataset.", e) from e
+    if response["http_status"] != HTTPStatus.OK:
+        raise PreviousStepStatusError(
+            f"Previous step gave an error: {response['http_status']}. This job should not have been created."
+        )
+    content = response["content"]
+
+    if "dataset_info" not in content:
+        raise PreviousStepFormatError("Previous step did not return the expected content.")
+    dataset_info = content["dataset_info"]
```

Comment:
these four lines should be removed if we depend on /dataset-info

Reply:
Done, thanks!